src/tui/editor.zig

```diff
@@ -3557,20 +3579,31 @@ pub const Editor = struct {
                     .col = 0,
                 },
             };
-            new_cursel.*.?.cursor.move_end(root, self.metrics);
+            if (with_selections) {
+                if (!is_eol_right(root, &new_cursel.*.?.cursor, self.metrics)) {
+                    try self.select_line_at_cursor(root, &new_cursel.*.?, .exclude_eol);
+                    new_cursel.*.?.cursor.target = new_cursel.*.?.cursor.col;
+                    new_cursel.*.?.selection.?.end.target = new_cursel.*.?.selection.?.end.col;
+                }
+            } else {
+                new_cursel.*.?.cursor.move_end(root, self.metrics);
+            }
         }
     }
 
-    pub fn add_cursors_to_line_ends(self: *Self, _: Context) Result {
+    pub fn add_cursors_to_line_ends(self: *Self, ctx: Context) Result {
         const root = try self.buf_root();
         const cursels = try self.cursels.toOwnedSlice(self.allocator);
         defer self.allocator.free(cursels);
+        var with_selections: bool = false;
```

Comment:
I would recommend you use an enum instead of bool. That is much easier to remember what it does in the keybindings file.

Something like `var mode: enum{ with_selections, normal } = .normal;`